### ISSUE TITLE
accessibilityRole="heading" seems to cause problems on Android

### ISSUE THREAD

**Issue by BogdanCerovac** [expo/expo#20712]
### Do you understand that any discussions or questions opened as issues in the core Expo repository will be closed?

Yes

**Comment by BogdanCerovac** [comment]
Please feel free to move this question to issues if it's more appropriate.

I've made a simple Snack example to test `accessibilityRole="heading"` with my Android phone (Samsung Galaxy S8 and Samsung Galaxy S22) and noticed that it seems to break rendering without reporting any issues.

I made a [video that demonstrates that normal text worked OK](https://youtu.be/MVd4WHjjkHA), but when I added the `accessibilityRole="heading"` to `<Text>` the whole app didn't render and was unable to refresh/update. Same issue when using virtual device with Android. Changing SDKs didn't help, updated Expo app on phones, didn't help...

iOS worked well. Web got correct semantics.

Seems that problem is directly connected to adding `accessibilityRole="heading"`. Console clean, no issues seen.

How can I debug this? Couldn't find similar issues...

Link to Snack: https://snack.expo.dev/agJkMBIPY

Code for reference;

`export default function App() {
  return (
    <View style={styles.container}>
      <Text accessibilityRole="heading" style={styles.heading}>
        Application 
      </Text>
     <Text style={styles.paragraph}>
        Lorem ipsum and so on...
      </Text>
    </View>
  );
}`
